test/e2e/scheduling.go

```diff
@@ -92,96 +96,162 @@ var _ = Describe("ReplicaSchedulingPreferences", func() {
 		}
 		namespace = f.TestNamespaceName()
 		if framework.TestContext.RunControllers() {
-			fixture := managed.NewSchedulerControllerFixture(tl, f.ControllerConfig())
-			f.RegisterFixture(fixture)
+			controllerFixture, controller = managed.NewSchedulerControllerFixture(tl, f.ControllerConfig())
+			f.RegisterFixture(controllerFixture)
 		}
 	})
 
-	testCases := map[string]struct {
-		total         int32
-		weight1       int64
-		weight2       int64
-		min1          int64
-		min2          int64
-		cluster1      int32
-		cluster2      int32
-		noPreferences bool
-	}{
-		"replicas spread equally in clusters with no explicit per cluster preferences": {
-			total:         int32(4),
-			cluster1:      int32(2),
-			cluster2:      int32(2),
-			noPreferences: true,
-		},
-		"replicas spread in proportion of weights when explicit preferences with weights specified": {
-			total:    int32(6),
-			weight1:  int64(2),
-			weight2:  int64(1),
-			min1:     int64(0),
-			min2:     int64(0),
-			cluster1: int32(4),
-			cluster2: int32(2),
-		},
-		"replicas spread considering min replicas when both minreplica and weights specified": {
-			total:    int32(6),
-			weight1:  int64(2),
-			weight2:  int64(1),
-			min1:     int64(3),
-			min2:     int64(3),
-			cluster1: int32(3),
-			cluster2: int32(3),
-		},
-	}
-
-	for key := range schedulingTypes {
-		typeConfigName := key
+	Describe("SchedulingManager", func() {
+		Context("when federatedtypeconfig resources are changed", func() {
+			It("related scheduler and plugin controllers should be dynamically disabled/enabled", func() {
+				if framework.TestContext.RunControllers() {
```

Comment:
My apologies for missing this earlier!  Please call `framework.Skip` if `RunControllers()` is `false` to ensure a skip is recorded instead of a pass.  It's also a general best practice to avoid unnecessary indentation where possible.

Reply:
Good catch. Thanks.
Resolved.